### ISSUE TITLE
JsonSchema enum's bug

### ISSUE THREAD

**Issue by jykj2020** [josdejong/svelte-jsoneditor#210]
Only the enum at layer 1 takes effect as select，I'm using vanilla-jsoneditor. 

scheme:
```json
{
  "schema": {
    "type": "object",
    "properties": {
      "pkey": {
        "title": "Table Pkey",
        "type": "string",
        "enum": ["t1","t2","t3"]
      },
      "fkey": {
        "title": "Table Fkey",
        "type": "array",
        "items": {
          "type": "object",
          "properties": {
            "key": {
              "title": "select key",
              "type": "string",
              "enum": ["col1","col2","col3"]
            },
            "table": {
              "title": "select table",
              "type": "string",
              "enum": ["t1","t2","t3"]
            }
          }
        }
      }
    }
  }
}
```

**Comment by josdejong** [comment]
Fixed via https://github.com/josdejong/svelte-jsoneditor/commit/887bf23c6aeb63de3a75b677f0545a51efc3c449 (not yet published).

The cause was that the `findSchema` function was relying on the path having array index values being a number, like `['fkey', 0, 'table']`, whereas the path in the editor was changed to contain only strings, like `['fkey', '0', 'table']` (closer to JSONPointer and solving ambiguity thoughout the code base). I've also converted the code into TypeScript which wasn't yet the case.
